### DISCUSSION TITLE
Cannot configure network after network conneciton is lost

### DISCUSSION THREAD

**Discussion by wailnucleus** [Hypfer/Valetudo#1794] [Q&A / Support]
Hi all

I've lost my robot's network and serial console and kindly asked for input [here](https://github.com/Hypfer/Valetudo/discussions/1780).

I can now reset the WiFi by pressing and holding the two outer buttons, but the Valetudo WLAN configuration always fails with an error message

What I've done so far:

Connect with robot AP
Login to Valetudo through http://192.168.5.1

Configure the WLAN in Valetudo WegGUI
Now I get the following error message from robot after some time (seems to be a timeout):
"Unable to connect to the server. Please check your local network and please try again..."

I can ssh to the robot and do some checks:

Robot:
[root@p2029_release:~]# ls -ltr /data/config/miio
total 2
-rw-r--r--    1 root     root           168 Jan  1 00:00 device.conf
-rw-r--r--    1 root     root            17 Jan  1 00:00 device.token

Backup I took some time ago:
ls -ltr data/config/miio/
total 20
-rw-r--r-- 1 oec oec 168 Jan  1  1970 device.conf
-rw-r--r-- 1 oec oec  17 Jan  1  1970 device.token
-rw-r--r-- 1 oec oec  52 Jan  1  1970 wifi.conf
-rw-r--r-- 1 oec oec   2 Jan  1  1970 device.uid
-rw-r--r-- 1 oec oec   3 Jan  1  1970 device.country

-> Conclusion: some files are missing

log entry:
"set empty to /data/config/ava/iot.flag"

[root@p2029_release:/data/log]# cat /data/config/ava/iot.flag
[root@p2029_release:/data/log]# 
-> iot flag is empty

In my backup, it is not empty:
cat config/ava/iot.flag 
miio

WLAN config looks good on the robot in /data/config/wifi/ap_config.json

I would like to try last part of https://builder.dontvacuum.me/dreame/cmds.txt, but of course, I do not want to brick my vacuum, especially as the serial console does not seem to work anymore:

rm -rf /data/config/miio/*
reboot
- wait for the open WiFi of the robot to show up
- press the Home and Spot button at the same time (if the open WiFi does not show up)
- connect to that WiFi
- navigate to http://192.168.5.1 in your browser
- go to settings/wifi and enter your WiFi credentials

Any comments or recommendations? 

Kind regards

**Comment by mandomal** [comment]
I don't have specific advice. How did you find the IP you needed to connect to?

**Reply by wailnucleus** [reply]
Thank you, mandomal, Valetudo is actually running on the robot. The network configuration seems to be broken and I am unable to re-confiure it. But I cannot easily debug it as the serial console does not work anymore.

When I reset the WLAN on the robot, I can connect to the robot's WIFI (dreame-vacuum-p2029_miapXXXX). I open http://192.168.5.1 on my browser. I can authenticate with the username/password I configured for Valetudo when I installed it the first time a year ago. I can also SSH to the robot's IP  as usual with my key when in "WLAN setup mode".
When actually I try to re-configure the WLAN now
![Screenshot_2023-04-19_20-22-02](https://user-images.githubusercontent.com/42916345/233167251-a6066bd3-79df-45bb-9616-4ea3a0dd2501.png)

by inserting the WLAN SSID and password, I get a timeout error from the robot after some time and the new configuration does not seem to be saved or activated on the robot.

When I reboot, the robot does not connect anymore to the home WIFI. However,  when I reset the robot's WIFI, what I can again reset the WLAN and start over..

The remaining issues for now are that when I set the SSID and PSK/Password in the Valetudo WebGUi, the config is not saved and that I somehow lost the serial console.

Hence I would like to try last part of https://builder.dontvacuum.me/dreame/cmds.txt, but of course, I do not want to brick my vacuum, especially as the serial console does not seem to work anymore:

"(If you used wifi_cli, you need to delete files in /data/config/miio, reboot, connect to the robot WiFi and provision the WiFi via Valetudo):
rm -rf /data/config/miio/*
reboot"
-> then hopefully the network will be back.

Thank you for your consideration.
